pyquil/resource_manager.py

```diff
@@ -116,7 +117,7 @@ def reset(self):
         """
         Frees all qubits, resetting this ResourceManager.
         """
-        for qubit in self.live_qubits:
+        for qubit in copy(self.live_qubits):
```

Comment:
Can you add a comment above this to explain why a copy is necessary? Something like

> A copy is necessary because as we traverse `self.live_qubits`, the list gets modified, which has consequences on the traversal itself.